server/src/test/java/org/apache/uniffle/server/MockedShuffleServerGrpcService.java

```diff
@@ -97,6 +97,14 @@ public void getShuffleResult(RssProtos.GetShuffleResultRequest request,
       LOG.info("Add a mocked timeout on getShuffleResult");
       Uninterruptibles.sleepUninterruptibly(mockedTimeout, TimeUnit.MILLISECONDS);
     }
+    if (numOfFailedReadRequest > 0) {
```

Comment:
After adding this code, the UT of RSSStageResubmitTest will fail because it does not support getShuffleResult and throws FetchFailed., we can reuse the test logic of RSSStageResubmitTest.